ProjFS.Mac/PrjFSKextTests/KauthHandlerTests.mm

```diff
@@ -58,4 +65,189 @@ - (void)testShouldIgnoreVnodeType {
+    shared_ptr<vnode> testVnodeHfs = vnode::Create(testMountHfs, "/hfs");
+    XCTAssertFalse(
+        ShouldHandleVnodeOpEvent(
+            &perfTracer,
+            context,
+            testVnodeHfs.get(),
+            action,
+            &vnodeType,
+            &vnodeFileFlags,
+            &pid,
+            procname,
+            &kauthResult,
+            &kauthError));
+    XCTAssertEqual(kauthResult, KAUTH_RESULT_DEFER);
+    
+    
+    // Test invalid VNODE Type
+    shared_ptr<vnode> testVnodeInvalidType = vnode::Create(testMount, "/foo2", VNON);
+    XCTAssertFalse(
+        ShouldHandleVnodeOpEvent(
+            &perfTracer,
+            context,
+            testVnodeInvalidType.get(),
+            action,
+            &vnodeType,
+            &vnodeFileFlags,
+            &pid,
+            procname,
+            &kauthResult,
+            &kauthError));
+    XCTAssertEqual(kauthResult, KAUTH_RESULT_DEFER);
+
+    
+    // Test failure reading attr
+    testVnode->errors.getattr = EBADF;
+    XCTAssertFalse(
+        ShouldHandleVnodeOpEvent(
+            &perfTracer,
+            context,
+            testVnode.get(),
+            action,
+            &vnodeType,
+            &vnodeFileFlags,
+            &pid,
+            procname,
+            &kauthResult,
+            &kauthError));
+    XCTAssertEqual(kauthResult, KAUTH_RESULT_DENY);
+    // reset to valid value
+    testVnode->errors.getattr = 0;
+
+    
+    // Test invalid file flag
+    testVnode->attrValues.va_flags = FileFlags_IsEmpty;
+    XCTAssertFalse(
+        ShouldHandleVnodeOpEvent(
+            &perfTracer,
+            context,
+            testVnode.get(),
+            action,
+            &vnodeType,
+            &vnodeFileFlags,
+            &pid,
+            procname,
+            &kauthResult,
+            &kauthError));
+    XCTAssertEqual(kauthResult, KAUTH_RESULT_DEFER);
+    // reset to valid value
+    testVnode->attrValues.va_flags = FileFlags_IsInVirtualizationRoot;
+    
+
+    // Test with file crawler trying to populate an empty file
+    testVnode->attrValues.va_flags = FileFlags_IsEmpty | FileFlags_IsInVirtualizationRoot;
+    char mds[] = "mds";
+    SetProcName(mds);
+    XCTAssertFalse(
+        ShouldHandleVnodeOpEvent(
+            &perfTracer,
+            context,
+            testVnode.get(),
+            action,
+            &vnodeType,
+            &vnodeFileFlags,
+            &pid,
+            procname,
+            &kauthResult,
+            &kauthError));
+    XCTAssertEqual(kauthResult, KAUTH_RESULT_DENY);
+
+    
+    // Test with finder trying to populate an empty file
+    char finder[] = "Finder";
+    SetProcName(finder);
+    XCTAssertTrue(
+        ShouldHandleVnodeOpEvent(
+            &perfTracer,
+            context,
+            testVnode.get(),
+            action,
+            &vnodeType,
+            &vnodeFileFlags,
+            &pid,
+            procname,
```

Comment:
How vigelent should we be abount freeing memory?
We aren't freeing procname here, but the memory will be returned when the test is over correct?

Reply:
`procname` is a local variable of array type, so it's allocated on the stack and automatically disappears when the function returns, so in this case there's nothing to worry about.

Reply:
I copied the pattern you had for Path.  Using string is a nice way to not have to worry about free-ing.